internal/cmdcommon/prompt/oneofstringlist.go

```diff
@@ -0,0 +1,72 @@
+package prompt
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"slices"
+	"strings"
+)
+
+type OneOfStringList struct {
+	reader     io.Reader
+	writer     io.Writer
+	message    string
+	promptText string
+	values     []string
+}
+
+func NewOneOfStringList(message, promptText string, values []string) *OneOfStringList {
+	return &OneOfStringList{
+		reader:     os.Stdin,
+		writer:     os.Stdout,
+		message:    message,
+		promptText: promptText,
+		values:     values,
+	}
+}
+
+func NewCustomOneOfStringList(reader io.Reader, writer io.Writer, message, promptText string, values []string) *OneOfStringList {
+	return &OneOfStringList{
+		reader:     reader,
+		writer:     writer,
+		message:    message,
+		promptText: promptText,
+		values:     values,
+	}
+}
+
+func (l *OneOfStringList) Prompt() (string, error) {
+	var userInput string
+	fmt.Fprintf(l.writer, "%s\n%s\n\n%s", l.message, l.valuesListString(), l.promptText)
+	_, err := fmt.Fscan(l.reader, &userInput)
+	if err != nil && err == io.EOF {
+		return "", fmt.Errorf("no value was selected")
+	}
```

Comment:
I think we should handle the case when the error is not equal to the `io.EOF` and then return it